tests/Unit/AdminDeleteTest.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Tests\Feature;
+
+use App\Models\User;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Tests\TestCase;
+
+class AdminDeleteTest extends TestCase
+{
+
+    public function test_admin_can_delete_users()
```

Comment:
Praticamente tudo ok. A única coisa é que o correto seria criarmos um usuário, para então, deleta-lo. Pois quando coloco um dd para debugar o test, por mais que retorne 204, não foi feita a requisição de fato.

Para criar esse user, podemos usar a factory só mudando o profile_id para 2,3 ou4;

Exemplo:
$recepcionista = User::factory()->create(['profile_id' => 2]);

e depois na hora da deleção podemos chamar assim:
$response = $this->withHeader('Authorization', 'Bearer ' . $token)->delete('/api/users/' . $recepcionista->id);

Isso vale para os demais testes também.